### ISSUE TITLE
Scan receives unexpected type []uint8 error

### ISSUE THREAD

**Issue by waltcow** [ent/ent#3019]
hello, my database type is  **mysql 8.0.24** , ent client connect with option `charset=utf8mb4&parseTime=True`

my ent schema as follow
```golang
func (SysUser) Fields() []ent.Field {
	return []ent.Field{
		field.Uint64("id"),
		field.JSON("role_ids", []uint64{}),	
                ...
	}
}
```

when i query in following way
```golang

var users []ent.SysUser

err := client.SysUser.Query().
		Where(sysuser.ID(1)).
		Modify(func(s *sql.Selector) {

		}).Scan(context.Background(), &users)
```

it failed with error 

`sql/scan: failed scanning rows: sql: Scan error on column index 12, name "role_ids": unsupported Scan, storing driver.Value type []uint8 into type *[]uint64`

but there are no problems when i query in normal way 

```golang
users, err := client.SysUser.Query().Where(sysuser.ID(1)).All(context.Background())
```

**Comment by a8m** [comment]
Hey @waltcow, the `Scan` API does not support JSON fields atm. We can support it, but is there any reason for not using `All` in this case?

Also, it's a duplicate of https://github.com/ent/ent/issues/2052.

**Comment by waltcow** [comment]
> 

@a8m  in my case, I just want to join tables and fetch extra fields and scan via nested struct like

```golang

type SysUserDetails {
     ent.SysUser
     RoleName string `json:role_name`
}
```

**Comment by a8m** [comment]
Hey! Issue was resolved with #3022. 

Please, let me know if you still have trouble with it.
